.ci/lint-python.sh

```diff
@@ -2,11 +2,9 @@
 
 set -e -E -u -o pipefail
 
-# this can be re-enabled when this is fixed:
-# https://github.com/tox-dev/filelock/issues/337
-# echo "running pre-commit checks"
-# pre-commit run --all-files || exit 1
-# echo "done running pre-commit checks"
+echo "running pre-commit checks"
+pre-commit run --all-files || exit 1
+echo "done running pre-commit checks"
```

Comment:
I had commented this out in https://github.com/microsoft/LightGBM/pull/6377#issuecomment-2164314791, to keep merging things in preparation for the release.

Looks like the issues with `filelock` have been resolved by a new release of that project, and this now runs successfully again 🎉